include/pybind11/pytypes.h

```diff
@@ -1091,7 +1091,7 @@ class str : public object {
 
 # ifdef PYBIND11_HAS_U8STRING
     // NOLINTNEXTLINE(google-explicit-constructor)
-    str(std::u8string_view s) : str(s.data(), s.size()) { }
+    str(std::u8string_view s) : str(reinterpret_cast<const char*>(s.data()), s.size()) { }
```

Comment:
Is there a cleaner way to obtain a `char*` from a `char8_t*`?
I look at `reinterpret_cast` as a method of last resort.
If there is no alternative, adding a comment would be good, to explain why this is sufficiently safe and portable.
E.g.: // The C++ standard guarantees that sizeof(char) and sizeof(char8_t) are identical.

Reply:
Looks like there is no alternative (sigh ... omg ... sigh).

https://stackoverflow.com/questions/57603013/how-to-safely-convert-const-char-to-const-char8-t-in-c20

Reply:
I've added a comment.

> Looks like there is no alternative (sigh ... omg ... sigh).

Yet another reason to dislike char8_t/u8string_view/etc.  It all feels a bit like a solution in search of an artificial problem that no one actually has *and* breaks code in the wild that actually thought it might be safe to use u"" to properly indicate utf-8 literals before C++20 (e.g. `std::string s = u8"asdf";` no longer compiles under -std=c++20).  Hurray.
\</rant\>